samples/cpp/intelligent_scissors.cpp

```diff
@@ -0,0 +1,210 @@
+#include <iostream>
+#include <cmath>
+#include <string>
+#include <vector>
+#include <queue>
+#include <opencv2/core/core.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+
+using namespace cv;
+
+Mat img, img_pre_render, img_render;
+Point end;
+std::vector<std::vector<Point> > contours(1);
+std::vector<Point> tmp_contour;
+Mat zero_crossing, gradient_magnitude, Ix, Iy, hit_map_x, hit_map_y;
```

Comment:
Using of global objects is a bad practice.
OpenCV samples should not recommend that.

Create class and wrap this stuff there.